### ISSUE TITLE
Niv's TODOs

### ISSUE THREAD

**Issue by yossigil** [SpartanRefactoring/Spartanizer#134]

**Comment by yossigil** [comment]
Anything I could do to help?

**Comment by NivShalmon** [comment]
I responded to a TODO in InfixPlusRemoveParenthesis in code, I added a TODO: Yossi there so you would see. If I'm correct than you can just remove the TODO after reading it

**Comment by yossigil** [comment]
:100:

**Comment by NivShalmon** [comment]
filling in reports for previous weeks, as well as this week.

**week 1**

- #1125 
- first implementation of #1150 (has been reopened since due to bugs)
- add examples to a bloater ([link to commit](https://github.com/SpartanRefactoring/Main/commit/85b08c5547f2f500e065b9fc74fa9d1bba15446e))

**week 2**

- wrote "Continuous Integration" wiki with @ravivos (#1194)
- #1146 
- #1129
- better unit testing-  #1172 

**week 3**

 - #187
 - #1225

**week 4**
 - #1204
 - work on #1257, led to the decision not to solve it for now

**week 5**
 - started work on #1225, drawing the buttons and setting basic listeners.

due to the holidays and #1310 preventing me from working on #1225 for most of this week, and since #1225 was my only issue for said week, not much work was done.

**Comment by NivShalmon** [comment]
**week 6**

- #1125 finished
- #1318 - the operation currently supports two of the three configurations it should support, the last one awaits some enhancment issues so put on hold rather than closed
- #116 fix small text issue

**Comment by NivShalmon** [comment]
**week 7**
Vacation #1154

**Comment by NivShalmon** [comment]
**week 8**

- #1350 
- #1161, while working on #1350 
- started work on #1321, not done yet due to bugs with serialization

**Comment by NivShalmon** [comment]
**week 9**
a lot of work since we were preparing for a release

- #1321 - a lot of work here to along with @ravivos to finally make the prefernces page and widget comunicate properly through the prefernces store
- #1419 
- #1354 and #1410 - similar bugs in the same method
- #1397

**Comment by NivShalmon** [comment]
**week 10**

- #408 fixed tests, improved type's lookdown to use namespace to find type of a variable, removed wrong tipper. Issue currently awaits rewriting of the tipper to use type.
- #888 improvments to type's lookup to properly handle varible declaration and infixExpressions. Currently awaits fix to the tipper.
- #1424 fix a bug in type
- #1421 created the bloater and basic tests, currently pending on #1453 for more tests
- #1407 with @dormaayan

**Comment by NivShalmon** [comment]
**week 11**

- #1479 
- #1449 
- fixed tests of #1421

**Comment by NivShalmon** [comment]
**week 12**
vacation

**week 13**

- #1163 
- #1421 added tests
- #1498 
- #1523
- work on the instalation manual wiki and other small bug fixes in preperation for the release

**Comment by yossigil** [comment]
:+1:
